walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,136 @@
-  }
+    private ArrayList<TimeRange> availableTimes = new ArrayList<>();
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        availableTimes.clear();
+
+        // If the request is too long, return empty ArrayList
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return availableTimes;
+        }
+
+        /* Any time is available during the start. Time ranges will 
+            be dynamically added and removed from the availableTimes 
+            array as the program loops through the events */
+        availableTimes.add(TimeRange.WHOLE_DAY);
+        
+        // If the set of attendees is empty, the entire day is available
+        if (request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty()) {
+            return availableTimes;
+        }
+
+        Collection<String> attendees = request.getAttendees();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+
+        // Boolean: mandatoryAttendee
+        Map<String, Boolean> attendeesMap = new HashMap<String, Boolean>();
+        for (String attendee : attendees) { 
+            attendeesMap.put(attendee, true); 
+        }
+        for (String attendee : optionalAttendees) { 
+            attendeesMap.put(attendee, false); 
+        }
+
+        System.out.println(attendeesMap.toString());
+
+        /* Events stored in Map to allow to dynamically remove events in loop (so as not to 
+           work on the same event that is shared by multiple attendees) */
+        Map<Event, Boolean> eventsMap = new HashMap<Event, Boolean>();
+        for (Event event : events) { 
+            eventsMap.put(event, true); 
+        }
+
+        Iterator eventsIterator; 
+        for (Map.Entry<String, Boolean> attendeeEntry : attendeesMap.entrySet()) {
+            String attendee = attendeeEntry.getKey();
+
+            eventsIterator = eventsMap.entrySet().iterator();
+            while(eventsIterator.hasNext()) {
+
+                Map.Entry pair = (Map.Entry) eventsIterator.next();
+                if ((Boolean) pair.getValue()) {
+                    Event event = (Event) pair.getKey();
+
+                    Collection<String> eventAttendees = event.getAttendees();
+                    if (eventAttendees.contains(attendee)) {
+                        TimeRange timeRange = event.getWhen();
+                        blockTimeRange(timeRange, request.getDuration(), attendeeEntry.getValue());
+
+                        if (attendeeEntry.getValue()) {
+                            eventsIterator.remove();
+                        }      
+                        // If there are only optional attendees & no times are available for them
+                        if (!attendeeEntry.getValue() && availableTimes.isEmpty()) {
+                            availableTimes.add(TimeRange.WHOLE_DAY);
+                        }
```

Comment:
I'm not sure I understand this part.  You've been building up your list of availableTimes in your previous loop, then this section you reset the work you've done so far by adding the whole day back in.  Then the loop continues on and blocks more time out of the list.